src/main/java/org/prebid/server/auction/LatestRatesService.java

```diff
@@ -0,0 +1,50 @@
+package org.prebid.server.auction;
+
+import io.vertx.core.http.HttpClient;
+import io.vertx.core.json.Json;
+import org.prebid.server.auction.model.Currency;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.util.Map;
+
+/**
+ * Class holding and updating latest currencies from server.
+ */
+public class LatestRatesService {
+
+    private static final String CURRENCY_SERVER_URL = "currency.prebid.org";
+    private static final Logger logger = LoggerFactory.getLogger(LatestRatesService.class);
+
+    private HttpClient httpClient;
+    private volatile Currency currency;
```

Comment:
While strictly speaking this doesn't make any harm I don't think it should be `volatile` - we really don't care if some event loop thread sees the updated reference not quite right after its update, the main thing is it will become visible _eventually_. Thus we can consciously trade thread safety for some performance gain here.